### PR TITLE
Feature/OH2-182 verify account with aws cognito

### PR DESCRIPTION
**Pull request by jokekerker**
- [x] OH2-237 welcome email
- [x] OH2-182 Verify account

### CODE REVIEW

**Review by sirawat21 — APPROVED** [review]

File: apps/mailer/src/mailer.service.ts [hunk 1]
```diff
@@ -1,8 +1,89 @@
...
+import { IServiceResponse } from '@app/rabbit';
+import { WelcomeMailerDto } from './dto/welcome.mailer.dto';
+import { SendMailerDto } from './dto/send.mailer.dto';
+import { HTTP_CODE_METADATA } from '@nestjs/common/constants';
 
 @Injectable()
-export class MailerService {
-  getHello(): string {
-    return 'Hello World!';
+export class MailerServiceExt {
```
Comment (ronsandova) [inline apps/mailer/src/mailer.service.ts:9]: Why is this call MailerServiceExt and not MailerService?

File: docs/db/email_verification.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE IF NOT EXISTS public.email_verifications
```
Comment (ronsandova) [inline docs/db/email_verification.sql:1]: maybe rename this table

Comment (ronsandova) [reply docs/db/email_verification.sql:1]: this table is missing account fk

File: libs/cognito/src/cognito.service.ts [hunk 3]
```diff
@@ -18,25 +19,28 @@ export class CognitoService {
...
+  ) {
+    const signUpResult = new Promise((resolve, reject) => {
       this.userPool.signUp(
         email,
         password,
         [],
         null,
         (err, result) => {
           if (!result) {
+            console.log('cognito ---> error ' + JSON.stringify(err));
```
Comment (ronsandova) [inline libs/cognito/src/cognito.service.ts:33]: Replace this for logger

File: libs/cognito/src/cognito.service.ts [hunk 4]
```diff
@@ -18,25 +19,28 @@ export class CognitoService {
...
             reject(err);
           } else {
+            console.log('cognito ---> success ' + JSON.stringify(result.user));
```
Comment (ronsandova) [inline libs/cognito/src/cognito.service.ts:36]: Replace this for logger

**Review by ronsandova — COMMENTED** [review]
A few changes please @jokekerker

File: apps/mailer/src/mailer.service.ts (hunk 1 above)
Comment (jokekerker) [reply apps/mailer/src/mailer.service.ts:9]: because class name is duplicate with @nestjs-modules/mailer

Comment (jokekerker) [reply apps/mailer/src/mailer.service.ts:10]: I see we need to use nodemailer instead of Mailer

Comment (ronsandova) [reply apps/mailer/src/mailer.service.ts:10]: I like sendgrid, but not sure what to do here

File: apps/account/src/services/account.service.ts [hunk 5]
```diff
@@ -109,38 +115,108 @@ export class AccountService {
...
+
+      const emailVerification = await this.emailVerificationRepository.findOneBy({ email: email });
+
+      const emailVerificationObj = {
+        email: email,
+        email_token: v4(),
+        created_at: new Date()
+      }
+
+      const sendMailObj = {
```
[the remark this answers is not in the document]
Comment (jokekerker) [reply apps/account/src/services/account.service.ts:141]: please provide me the template I can add into sendgrid

**Review by ronsandova — APPROVED** [review]

**Pull request merged**